web/src/hooks/useApps.ts

```diff
@@ -165,44 +324,90 @@ const deleteAppFetcher = async (
 ) => {
   const { id } = args.arg;
   const response = await graphQLRequest<{
-    app: Pick<AppModel, "id">;
+    delete_app_by_pk: string;
   }>({
     query: DeleteAppQuery,
     variables: { id },
   });
-  if (response.data?.app) {
-    return response.data?.app;
+  console.log(response);
+  if (response.data?.delete_app_by_pk) {
+    return response.data?.delete_app_by_pk;
   }
   throw Error("Could not delete app");
 };
 
-type NewAppPayload = Pick<
-  AppModel,
-  "name" | "description_internal" | "engine" | "is_staging"
->;
+type NewAppPayload = Pick<AppModel, "engine" | "is_staging"> & {
+  app_metadata: NewAppMetadataPayload;
+};
+type NewAppMetadataPayload = Pick<AppMetadataModel, "name" | "description">;
 
 const insertAppFetcher = async (_key: string, args: { arg: NewAppPayload }) => {
-  const { name, description_internal, engine, is_staging } = args.arg;
-
-  const response = await graphQLRequest<{
+  const { engine, is_staging, app_metadata } = args.arg;
+  if (!app_metadata) {
+    throw new Error("No app metadata provided");
+  }
+  const appResponse = await graphQLRequest<{
     insert_app_one: AppModel;
   }>({
     query: InsertAppQuery,
     variables: {
-      object: {
-        name,
-        description_internal,
+      appObject: {
```

Comment:
wondering if we can insert both in a single query

Reply:
cannot we need the id from the first call and apollo doesn't let you chain that way